src/components/shared/Forms/Collection/BaseCollectionForm.js

```diff
@@ -200,8 +208,9 @@ class BaseCollectionForm extends Component {
       return;
     }
 
-    const properties = cleanProperties(this.getPropertiesFromState());
-    const i18n       = this.cleanI18n();
+    const properties  = cleanProperties(this.getPropertiesFromState());
+    const i18n        = this.cleanI18n();
+    properties.source = this.state.source || '';
```

Comment:
no need to have source as key in all properties on all collections.
We only need it in specific types of collection.

Reply:
add If